pkg/actuators/machine/validations.go

```diff
@@ -0,0 +1,169 @@
+package machine
+
+import (
+	"fmt"
+	ovirtconfigv1 "github.com/openshift/cluster-api-provider-ovirt/pkg/apis/ovirtprovider/v1beta1"
+	ovirtC "github.com/openshift/cluster-api-provider-ovirt/pkg/clients/ovirt"
+	apierrors "github.com/openshift/machine-api-operator/pkg/controller/machine"
+	ovirtsdk "github.com/ovirt/go-ovirt"
+)
+
+// validateMachine validates the machine object yaml fields and
+// returns InvalidMachineConfiguration in case the validation failed
+func validateMachine(ovirtClient ovirtC.OvirtClient, config *ovirtconfigv1.OvirtMachineProviderSpec) *apierrors.MachineError {
+	// UserDataSecret
+	if config.UserDataSecret == nil {
+		return apierrors.InvalidMachineConfiguration(
+			fmt.Sprintf("%s UserDataSecret must be provided!", ErrorInvalidMachineObject))
+	} else if config.UserDataSecret.Name == "" {
+		return apierrors.InvalidMachineConfiguration(
+			fmt.Sprintf("%s UserDataSecret *Name* must be provided!", ErrorInvalidMachineObject))
+	}
+
+	err := validateInstanceID(config)
+	if err != nil {
+		return err
```

Comment:
Wrapping errors...

Reply:
Edited the validateInstanceID function to return an error an wrapped it in the main function with apierrors.InvalidMachineConfiguration